### PR TITLE
Configurable SSH key exchange algorithm and MAC suite

### PR DESCRIPTION
**Pull request by virvum**
Add new configuration keys `SSH_SERVER_KEY_EXCHANGES` and `SSH_SERVER_MACS` in order to configure key exchange algorithm and MAC suite.

### CODE REVIEW

**Review by lafriks — CHANGES_REQUESTED** [review]
Please add also defaults in code like for chippers here: https://github.com/cumgun/gitea/blob/33711b6a963ee5fc5b836acc6cc1c9f577493a16/modules/setting/setting.go#L114

**Comment by virvum** [comment]
@lafriks done

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
LGTM

File: modules/setting/setting.go [hunk 1]
```diff
@@ -732,6 +736,14 @@ func NewContext() {
 	if len(serverCiphers) > 0 {
 		SSH.ServerCiphers = serverCiphers
 	}
+	serverKeyExchanges := sec.Key("SSH_SERVER_KEY_EXCHANGES").Strings(",")
```
Comment (sapk) [inline modules/setting/setting.go:739]:
~~~Shouldn't this be a default empty string ? ( sec.Key("SSH_SERVER_KEY_EXCHANGES").Strings("") )~~~
~~~For example if SSH_SERVER_KEY_EXCHANGES is not set. serverKeyExchanges = "," -> len(serverKeyExchanges) > 0 == true~~~

**Review by sapk — CHANGES_REQUESTED** [review]
~~~This will lead to a default list empty.~~~

File: modules/setting/setting.go (hunk 1 above)
Comment (sapk) [reply modules/setting/setting.go:739]: Sorry misread ... the .String**s**

**Review by sapk — APPROVED** [review]

**Comment by sapk** [comment]
LGTM

**Pull request merged**